### PR TITLE
Ubuntu1804

### LINKED ISSUE #897: Upgrade to Ubuntu 18.04 when it comes out

### PR DESCRIPTION
**Pull request by jackivanov**
- Fixes #897 #944 #956

Work in progress. Lightsail is not ready for Ubuntu 18.04 yet

- [x] DigitalOcean
~~- [ ] Amazon Lightsail~~
- [x] Amazon EC2
- [x] Microsoft Azure
- [x] Google Compute Engine
- [x] Scaleway
- [x] OpenStack (DreamCompute optimised)

### CODE REVIEW

**Comment by bytemare** [comment]
Have you tried this out for Scaleway successfully ?

When i put
`size: START1-S
image: Ubuntu Bionic Beaver`
in confi.cfg, I have the following error 
`TASK [cloud-scaleway : Create a server] ***************************************************************************************************************************************************************************
task path: /path/to/algo/roles/cloud-scaleway/tasks/main.yml:57
fatal: [localhost]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: 'image_id' is undefined\n\nThe error appears to have been in '/path/to/algo/roles/cloud-scaleway/tasks/main.yml': line 57, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Create a server\n      ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'image_id' is undefined"
}
Read vars_file 'config.cfg'`

But still works for Xenial.

**Comment by jackivanov** [comment]
@bytemare Yes, it worked. Could you file a separate issue, please?

**Comment by jackivanov** [comment]
Sorry, don't need to. They've changed API a bit, there is pagination now. I will fix it

**Comment by dguido** [comment]
Hey all, we're temporarily killing support for Lightsail since they don't support Ubuntu 18.04 yet. I'd rather merge this PR since it solves a bunch of underlying issues with Algo than wait for Lightsail to catch up to every other cloud provider.

File: roles/dns_encryption/tasks/ubuntu.yml [hunk 1]
```diff
@@ -35,14 +35,12 @@
...
-#      [Service]
-#      MemoryLimit=16777216
-#      CPUAccounting=true
-#      CPUQuota=5%
-#  notify:
-#    - daemon-reload
-#    - restart dnscrypt-proxy
+- name: Ubuntu | Add capabilities to bind ports
+  copy:
+    dest: /etc/systemd/system/dnscrypt-proxy.service.d/99-capabilitie.conf
```
Comment (jplock) [inline roles/dns_encryption/tasks/ubuntu.yml]: spelling: `99-capabilities.conf`

File: playbooks/ubuntu.yml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 ---
 
 - name: Ubuntu | Install prerequisites
-  raw: sleep 10 && sudo apt-get update -qq && sudo apt-get install -qq -y python2.7
+  raw: "{{ item }}"
```
Comment (jplock) [inline playbooks/ubuntu.yml]: could this be:

```
- name: Ubuntu | Install prerequisites
  apt: name={{ item }} state=latest update_cache=yes
  with_items:
    - python2.7
    - sudo
```

Comment (jackivanov) [reply playbooks/ubuntu.yml]: Not really. Ubuntu supplies with python3 by default and we don't want to take such risk. Let me try to change `ansible_python_interpreter` to python3 for these particular tasks.

**Pull request merged**